provider/docker.go

```diff
@@ -29,38 +29,30 @@ const DockerAPIVersion string = "1.21"
 
 // Docker holds configurations of the Docker provider.
 type Docker struct {
-	BaseProvider
+	BaseProvider     `mapstructure:",squash"`
 	Endpoint         string     `description:"Docker server endpoint. Can be a tcp or a unix socket endpoint"`
 	Domain           string     `description:"Default domain used"`
-	TLS              *DockerTLS `description:"Enable Docker TLS support"`
+	TLS              *ClientTLS `description:"Enable Docker TLS support"`
 	ExposedByDefault bool       `description:"Expose containers by default"`
 }
 
-// DockerTLS holds TLS specific configurations
-type DockerTLS struct {
-	CA                 string `description:"TLS CA"`
-	Cert               string `description:"TLS cert"`
-	Key                string `description:"TLS key"`
-	InsecureSkipVerify bool   `description:"TLS insecure skip verify"`
-}
-
 func (provider *Docker) createClient() (client.APIClient, error) {
 	var httpClient *http.Client
 	httpHeaders := map[string]string{
 		// FIXME(vdemeester) use version here O:)
```

Comment:
ah :smile_cat:.
Basic idea is to put the version of `traefik` in the http header (user-agent) so that we can debug easier looking at docker logs :stuck_out_tongue_closed_eyes:.

This should be done, but not really related to this PR 👼